### PR TITLE
[#45] User Feedback Feature

### PR DESCRIPTION
**Pull request by Samuel-Campbell**
[#45]

- [x] Adds ability for users to submit feedback

### CODE REVIEW

**Comment by Samuel-Campbell** [comment]
screenshot from 2017-12-21 09-17-32
screenshot from 2017-12-21 09-17-46
screenshot from 2017-12-21 09-17-53

**Comment by arekmano** [comment]
UI LGTM

File: src/postgresql_db/models.py [hunk 1]
```diff
@@ -154,6 +154,13 @@ class File(db.Model):
     timestamp = db.Column(db.DateTime, default=datetime.datetime.utcnow)
 
 
+class Feedback(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+
+    # Attributes
+    feedback = db.Column(db.String(250), nullable=False)
```
Comment (Vynny) [inline src/postgresql_db/models.py]: This should be db.Text() if we aren't imposing length limits.

File: src/web_client/src/components/Sidebar.vue [hunk 2]
```diff
@@ -82,13 +97,32 @@ export default {
...
+            dialogVisible: false,
             username: this.$localStorage.get('username').toUpperCase(),
             //TODO: fetch username from conversation, now use usertype instead
             api_url: process.env.API_URL,
             connectionError: false
         }
     },
     methods: {
+        submitFeedback(){
+            var feedback = document.getElementById("feedback-text").value;
```
Comment (Vynny) [inline src/web_client/src/components/Sidebar.vue]: This does not leverage vue.js. You should do this by either passing the value into the function on submit, or add a feedback object to 'data' then bind it to the input with v-model. Then in the method you can do 'this.feedback.text' to retrieve the text.

File: src/web_client/src/components/Sidebar.vue [hunk 3]
```diff
@@ -82,13 +97,32 @@ export default {
...
+            this.$http.post(this.api_url + 'feedback',{
+                data: feedback
+            }).then(
+                response => {
+                    alert("Thank you for providing us with feedback!");
+                    this.dialogVisible = false;
+                    document.getElementById("feedback-text").value = "";
```
Comment (Vynny) [inline src/web_client/src/components/Sidebar.vue]: Same as above, this should be bound to a model and not done with plain js.

**Review by Vynny — CHANGES_REQUESTED** [review]

File: src/web_client/src/components/Sidebar.vue (hunk 2 above)
Comment (Vynny) [reply src/web_client/src/components/Sidebar.vue]: You can see an example of this here: 
https://stackoverflow.com/a/45840995
https://vuejs.org/v2/guide/forms.html

File: src/web_client/src/components/Sidebar.vue (hunk 3 above)
Comment (arekmano) [reply src/web_client/src/components/Sidebar.vue]: +1

File: src/web_client/src/components/Sidebar.vue [hunk 4]
```diff
@@ -82,13 +97,32 @@ export default {
             uploadedFileList: new Array,
             openFileList: false,
             openReportList: false,
+            dialogVisible: false,
```
Comment (dodoels) [inline src/web_client/src/components/Sidebar.vue]: `openFeedbackModal` for consistent naming convention (start with verb)

File: src/web_client/src/components/Sidebar.vue [hunk 5]
```diff
@@ -61,6 +61,21 @@
                 <h3>PREVIOUS CASES</h3>
             </div>
             <!-- End of Previous Case List -->
+            <!-- Feedback -->
+            <div id="sidebar-feedback" class="sidebar-menu" @click="dialogVisible = true">
+                <h3>FEEDBACK</h3>
+            </div>
+            <!-- el-dialog for feedback -->
+            <el-dialog title="Feedback" :visible.sync="dialogVisible">
+                <textarea id="feedback-text">
```
Comment (dodoels) [inline src/web_client/src/components/Sidebar.vue]: 1. in `data ()`, in `user` object, create new value `feedback` 
2. add `v-model="user.feedback"`

File: src/web_client/src/components/Sidebar.vue [hunk 6]
```diff
@@ -82,13 +97,32 @@ export default {
...
             //TODO: fetch username from conversation, now use usertype instead
             api_url: process.env.API_URL,
             connectionError: false
         }
     },
     methods: {
+        submitFeedback(){
+            var feedback = document.getElementById("feedback-text").value;
+            this.$http.post(this.api_url + 'feedback',{
+                data: feedback
```
Comment (dodoels) [inline src/web_client/src/components/Sidebar.vue]: 3. `data: this.user.feedback`

File: src/web_client/src/components/Sidebar.vue (hunk 3 above)
Comment (dodoels) [reply src/web_client/src/components/Sidebar.vue]: 4. `this.user.feedback == null`

**Review by Vynny — APPROVED** [review]

**Review by dodoels — APPROVED** [review]
LGTM, I will add unit test and improve the code a bit later

**Pull request merged**